### PR TITLE
Remove const initializers from graph definition for compact metagraph; Also save model.pb

### PR DESCRIPTION
**Pull request by sjain-stanford**
Hi @jinyu121,
Thanks for the neat utility. I like it more than [darkflow](https://github.com/thtrieu/darkflow), mainly because you use TF-Slim API; hence cleaner graph scopes and simpler implementations of some layers such as Batch Normalization (avoids control flow and switch/cond nodes).

This PR contains changes I've made to support the following:
1. Replace const initializers (for pre-trained weights and BN params) with default initializers (random_normal / zeros etc.) when saving metagraph. This avoids duplication of weights and the resulting `.meta` is much more compact / portable.
1. Save graph definition to a protocol buffer `.pb`, alongside `.meta` and `.ckpt` previously generated.

Here's a comparison of size in bytes for yolov2 converted using DW2TF.

**Before (.meta = 195 MB):**
```
      2726  yolov2.cfg
 203934260  yolov2.weights
 204126037  yolov2.ckpt.meta   <==== (195MB)
 203892900  yolov2.ckpt.data-00000-of-00001
      3753  yolov2.ckpt.index
        79  checkpoint
```
**After (.meta = 0.3 MB):**
```
      2726  yolov2.cfg
 203934260  yolov2.weights
    313041  yolov2.ckpt.meta   <==== (0.3 MB)
 203892900  yolov2.ckpt.data-00000-of-00001
      3753  yolov2.ckpt.index
        79  checkpoint
```

Note that the pre-trained weights are still available as a `.ckpt`, hence this PR shouldn't impact regular graph execution. In fact it allows the user to train either from scratch (random init) or load from the pre-trained checkpoint.

As a secondary check, I've also validated the changes using randomized data for yolo-v2 and the input-output data match as expected.

Let me know your thoughts.

### CODE REVIEW

**Comment by jinyu121** [comment]
Thank you sooooooo much!
